File: backend/src/main/java/com/eagletech/ecommerce/backend/domain/model/Order.java
package com.eagletech.ecommerce.backend.domain.model;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import lombok.Data;

@Data
public class Order {
    private Integer id;
    private LocalDateTime dateCreated;
    private List<OrderProduct> orderProducts;
    private OrderState orderState;
    private User user;
    private Double total; // Campo para el total calculado

    public Order() {
        orderProducts = new ArrayList<>();
    }

    // Este método es útil para calcular el total sobre la marcha si es necesario, lo mantenemos.
    public BigDecimal getTotalOrderPrice(){
        return this.orderProducts.stream()
                .map(OrderProduct::getTotalItem)
                .reduce(BigDecimal.ZERO, BigDecimal::add);
    }
}

File: frontend/src/app/common/order.ts
import { OrderProduct } from "./order-product";
import { OrderState } from "./order-state";
import { User } from "./user";

export class Order {
    id: number | null;
    dateCreated: Date;
    orderProducts: OrderProduct[];
    user: User; // Reemplazado userId por el objeto User completo
    orderState: OrderState;
    total: number;

    constructor(
        id: number | null,
        dateCreated: Date,
        orderProducts: OrderProduct[],
        user: User,
        orderState: OrderState,
        total: number
    ) {
        this.id = id;
        this.dateCreated = dateCreated;
        this.orderProducts = orderProducts;
        this.user = user;
        this.orderState = orderState;
        this.total = total;
    }
}
